src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -921,14 +921,34 @@ def _resolve_policy_id(policy, policy_set_definition, client):
     policy_id = policy or policy_set_definition
     if not is_valid_resource_id(policy_id):
         if policy:
-            policy_def = client.policy_definitions.get(policy)
+            policy_def = _get_custom_or_builtin_policy(client, policy)
             policy_id = policy_def.id
         else:
-            policy_set_def = client.policy_set_definitions.get(policy_set_definition)
+            policy_set_def = _get_custom_or_builtin_policy(client, policy_set_definition, 'setdefinition')
             policy_id = policy_set_def.id
     return policy_id
 
 
+def _get_custom_or_builtin_policy(client, name, resourcetype='definition'):
+    if resourcetype != 'definition':
+        if supported_api_version(ResourceType.MGMT_RESOURCE_POLICY, min_api='2017-06-01-preview'):
+            errorObject = get_sdk(ResourceType.MGMT_RESOURCE_POLICY, 'ErrorResponseException', mod='models')
+            try:
+                return client.policy_set_definitions.get(name)
+            except errorObject as ex:
+                if ex.response.status_code == 404:
+                    return client.policy_set_definitions.get_built_in(name)
+        raise CLIError('policy set definitions are only supported with api-versions '
+                       '2017-06-01-preview and above')
+
+    from msrestazure.azure_exceptions import CloudError
+    try:
+        return client.policy_definitions.get(name)
+    except CloudError as ex:
+        if ex.status_code == 404:
+            return client.policy_definitions.get_built_in(name)
```

Comment:
Same as above. But here the exception is swallowed. Is it expected?

Reply:
Yes, this is what we want.

Reply:
The method ends here, it will returns `None` is it also expected?

Reply:
I'm now raising an exception if error code returned is other than 404